code/game/turfs/simulated/floor/chasm.dm

```diff
@@ -294,3 +295,30 @@
 	drop_y = y
 	var/list/target_z = levels_by_trait(SPAWN_RUINS)
 	drop_z = pick(target_z)
+
+/turf/simulated/floor/chasm/space_ruin
+
+/turf/simulated/floor/chasm/space_ruin/drop(atom/movable/AM)
+	//Make sure the item is still there after our sleep
+	if(!AM || QDELETED(AM))
+		return
+	falling_atoms[AM] = TRUE
+
+	var/list/target_z = levels_by_trait(SPAWN_RUINS) // list of all z levels that can get ruins spawned in
+	target_z -= AM.z // excluding the one atom was already in from possible z levels
+	var/turf/T = locate(rand(5, world.maxx - 5), rand(5, world.maxy - 5), pick(target_z))
+
+	if(T)
```

Comment:
can you make sure that players will only fall onto space turfs and not into the middle of another ruin?

Reply:
would sometime like this work without returning it if it doesn't pass the check?

```dm
if(T && istype(get_area(T), /area/space))
```